kfdef/kfctl_k8s_istio_dex.yaml

```diff
@@ -0,0 +1,387 @@
+      parameters:
+      - name: namespace
+        value: istio-system
+      repoRef:
+        name: manifests
+        path: istio/istio-crds
+    name: istio-crds
+  - kustomizeConfig:
+      parameters:
+      - name: namespace
+        value: istio-system
+      repoRef:
+        name: manifests
+        path: istio/istio-install
+    name: istio-install
+  # This component is the istio resources for Kubeflow (e.g. gateway), not about installing istio.
+  - kustomizeConfig:
+      overlays:
+      - https-gateway
+      parameters:
+        - name: clusterRbacConfig
+          value: "ON"
+      repoRef:
+        name: manifests
+        path: istio/istio
+    name: istio
+  - kustomizeConfig:
+      parameters:
+        - name: namespace
+          value: cert-manager
+      repoRef:
+        name: manifests
+        path: cert-manager/cert-manager-crds
+    name: cert-manager-crds
+  - kustomizeConfig:
+      parameters:
+        - name: namespace
+          value: kube-system
+      repoRef:
+        name: manifests
+        path: cert-manager/cert-manager-kube-system-resources
+    name: cert-manager-kube-system-resources
+  - kustomizeConfig:
+      overlays:
+        - self-signed
+        - application
+      parameters:
+        - name: namespace
+          value: cert-manager
+      repoRef:
+        name: manifests
+        path: cert-manager/cert-manager
+    name: cert-manager
+  - kustomizeConfig:
+      parameters:
+        - name: namespace
+          value: istio-system
+        - name: domain
+          value: DOMAIN #Replace with your kubeflow domain
+      repoRef:
+        name: manifests
+        path: istio/ingressgateway-self-signed-cert
+    name: ingressgateway-self-signed-cert
+  - kustomizeConfig:
+      overlays:
+        - application
+      parameters:
+        - name: namespace
+          value: istio-system
+        - name: userid-header
+          value: kubeflow-userid
+        - name: oidc_provider
+          value: http://dex.auth.svc.cluster.local:5556/dex
+        - name: oidc_redirect_uri
+          value: /login/oidc
+        - name: oidc_auth_url
+          value: /dex/auth
+        - name: skip_auth_uri
+          value: /dex
+        - name: client_id
+          value: kubeflow-oidc-authservice
+      repoRef:
+        name: manifests
+        path: istio/oidc-authservice
+    name: oidc-authservice
+  - kustomizeConfig:
+      overlays:
+        - istio
+      parameters:
+        - name: namespace
+          value: auth
+        - name: issuer
+          value: http://dex.auth.svc.cluster.local:5556/dex
+        - name: client_id
+          value: kubeflow-oidc-authservice
+        - name: oidc_redirect_uris
+          value: '["/login/oidc"]'
+        - name: static_email
+          value: admin@kubeflow.org
+        # Password is "password", 12-round bcrypt-hashed.
+        - name: static_password_hash
+          value: $2y$12$9Ukc61M.v83Nidxvcln9VeOtJRBBulSyH81hfnHzKAc53uZafn.6O
```

Comment:
Can this be an environment variable as well?

Reply:
+1 
password should be set via plugin based on environment variable